dynamic_programming/max_sub_array.py

```diff
@@ -37,7 +37,26 @@ def find_max_cross_sum(A,low,mid,high):
             right_sum=summ
             max_right=i
     return max_left,max_right,(left_sum+right_sum)
-    
+
+def max_sub_array(nums: List[int]) -> int:
+    """
+     Finds the contiguous subarray (can be empty array) 
+     which has the largest sum and return its sum.
+
+    >>> max_sub_array([-2,1,-3,4,-1,2,1,-5,4])
+    6
+    >>> max_sub_array([])
+    0
+    >>> max_sub_array([-1,-2.-3])
```

Comment:
Did you want a comma here instead of a period?

Reply:
yes, astonished why my local build did not catch that lol

Reply:
I am astonished by the results of __python3 -c "print(-2.-3)"__